### ISSUE TITLE
Perceptron Learner vs Logisitic Regression

### ISSUE THREAD

**Issue by theSparta** [aimacode/aima-python#282]
In the file `learning.py`, the `PerceptronLearner` function should not be same as Logisitic Regression.   There should be separate function for `PerceptronLearner` which uses **hard thresholding** as compared to **Logisitic regression** which uses the logistic function for threshold.

**Comment by MrDupin** [comment]
Hi theSparta.

Just checked, and you are correct. The PerceptronLearner function uses an activation function but in most of the literature I can remember, it doesn't.

I'm not sure how to proceed about this though. The issue is easily fixed, but maybe it was intended to be this way. Do you know who can we contact to get the green light to change this?

**Comment by theSparta** [comment]
Yeah, I also think that the `PerceptronLearner` intentionally uses the sigmoid function.

**Comment by satyarth934** [comment]
`def PerceptronLearner(dataset, learning_rate=0.01, epoches=100):
    """Logistic Regression, NO hidden layer"""
    i_units = len(dataset.inputs)
    o_units = 1  # As of now, dataset.target gives only one index.
    hidden_layer_sizes = []
    raw_net = network(i_units, hidden_layer_sizes, o_units)
    learned_net = BackPropagationLearner(dataset, raw_net, learning_rate, epoches)

    def predict(example):
        # Input nodes
        i_nodes = learned_net[0]

        # Activate input layer
        for v, n in zip(example, i_nodes):
            n.value = v

        # Forward pass
        for layer in learned_net[1:]:
            for node in layer:
                inc = [n.value for n in node.inputs]
                in_val = dotproduct(inc, node.weights)
                node.value = node.activation(in_val)

        # Hypothesis
        o_nodes = learned_net[-1]
        pred = [o_nodes[i].value for i in range(o_units)]
        return 1 if pred[0] >= 0.5 else 0

    return predict`

The comment in the second line states "Logistic Regression, NO hidden layer"

So I guess it is meant to be this way.
